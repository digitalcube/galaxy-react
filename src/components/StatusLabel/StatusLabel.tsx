import React, { FC } from 'react';
import { useMemo } from 'react';

export type StatusLabelProps = {
  isActive?: boolean;
  activeLabel?: string;
  inactiveLabel?: string;
  className?: string;
  fontWeight?: 'bold' | 'normal' | 'light';
};
export const StatusLabel: FC<StatusLabelProps> = ({
  className,
  isActive,
  fontWeight,
  activeLabel = 'on',
  inactiveLabel = 'off',
}) => {
  const classNames = useMemo(() => {
    return [
      'c-toggle-switch-status',
      isActive ? 'active' : 'inactive',
      `font-weight-${fontWeight}`,
      className,
    ].filter(Boolean);
  }, [className, isActive, fontWeight]);
  if (isActive === undefined) return null;
  return (
    <span className={classNames.join(' ')}>
      {isActive ? activeLabel : inactiveLabel}
    </span>
  );
};
