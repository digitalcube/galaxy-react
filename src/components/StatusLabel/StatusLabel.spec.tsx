import { render } from '@testing-library/react';
import React from 'react';
import { Default as StatusLabel } from './StatusLabel.stories';

describe('StatusLabel', () => {
  it('to match snapshot(default)', () => {
    const { asFragment } = render(
      <StatusLabel {...(StatusLabel.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
