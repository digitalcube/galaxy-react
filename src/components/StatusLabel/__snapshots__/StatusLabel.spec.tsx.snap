// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusLabel to match snapshot(default) 1`] = `
<DocumentFragment>
  <span
    class="c-toggle-switch-status active font-weight-bold"
  >
    on
  </span>
</DocumentFragment>
`;
